python/tchannel/tornado/peer.py

```diff
@@ -525,160 +525,123 @@ def send(self, arg1, arg2, arg3,
-                break
-            except ProtocolError as protocol_error:
-                # event: after_receive_protocol_error_per_attempt
-                self.peer.tchannel.event_emitter.fire(
-                    EventType.after_receive_system_error_per_attempt,
-                    request,
-                    protocol_error,
+                connection = yield peer.connect()
+                request = Request(
+                    service=self.service,
+                    argstreams=[
+                        InMemStream(endpoint), arg2.clone(), arg3.clone()
+                    ],
+                    id=connection.next_message_id(),
+                    headers=headers,
+                    endpoint=endpoint,
+                    ttl=ttl,
+                    tracing=Trace(
+                        name=endpoint,
+                        trace_id=trace_id,
+                        parent_span_id=parent_span_id,
+                        endpoint=Endpoint(peer.host,
+                                          peer.port,
+                                          self.service),
+                        traceflags=traceflag,
+                    )
                 )
 
-                (peer, connection) = yield self.prepare_for_retry(
-                    request, connection, protocol_error,
-                    peer, blacklist, retry_delay,
-                    num_of_attempt, attempt_times,
+                # event: send_request
+                self.peer.tchannel.event_emitter.fire(
+                    EventType.before_send_request, request,
                 )
+                response = yield self._send(connection, request)
 
-                if not connection:
-                    raise
-            except TimeoutError as operational_error:
-                # event: on_operational_error_per_attempt
+                # event: receive_response
                 self.peer.tchannel.event_emitter.fire(
-                    EventType.on_operational_error_per_attempt,
-                    request,
-                    operational_error,
+                    EventType.after_receive_response, request, response,
                 )
-
-                (peer, connection) = yield self.prepare_for_retry(
-                    request, connection, operational_error,
-                    peer, blacklist, retry_delay,
+                break
+            except (ProtocolError, TimeoutError) as error:
+                # Error Response will have tracing field.
+                if hasattr(error, "tracing"):
+                    # event: after_receive_error_response
+                    self.peer.tchannel.event_emitter.fire(
+                        EventType.after_receive_error_response, request, error,
+                    )
+                else:
+                    # event: on_outbound_error
+                    self.peer.tchannel.event_emitter.fire(
+                        EventType.on_outbound_error, request, error,
```

Comment:
I don't get the name of this event. The description says it's an operational error.